### PR TITLE
Allow setting platform macro settings externally

### LINKED ISSUE #2578: rpm/rpmbuild vs Yocto BSP platform names
Yocto is a cross-compiler system for building custom Linux based OSes.

The platform names in Yocto are more specific than basic CPU architecture names. They are called Board Support Packages (BSPs) and the build is  customized for such a specific SoC board, e.g. on ARM, MIPS or other exotic CPUs. Even when building for an Intel compatible device, the BSP name is e.g. genericx86 or genericx86_64, or when using the meta-intel layer, it can be e.g. intel-corei7-64, intel_skylake_64 and others, and these names end up in `/etc/rpm/platform`.

The problem is that the corresponding subdirectory does not get created in `/usr/lib/rpm/platform` so `rpmbuild` fails to run on a Yocto based system with all the necessary SDK present to build an RPM package from sources.

### PR DESCRIPTION
**Pull request by zboszor**
By default, rpm installs a series of default platforms based on the CPU architecture names in subdirectories called `/usr/lib/platform/<arch>-linux`.

This is enough for regular Linux distributions. However, some distributions may use more specific platform names that refer to particular computer systems, like SBCs or specific CPU tuning when compiling.

If the platform subdirectory does not exist in `/usr/lib/platform` then rpmbuild does not work.

Allow creating such custom platform subdirectory with feeding the necessary data using external variables: `RPM_CUSTOM_ARCH`, `RPM_CUSTOM_ISANAME`, `RPM_CUSTOM_ISABITS`, `RPM_CUSTOM_CANONARCH` and `RPM_CUSTOM_CANONCOLOR`

Fixes #2578

### CODE REVIEW

File: INSTALL [hunk 1]
```diff
@@ -143,6 +143,45 @@ and then install with:
...
 
+By default, rpm installs a series of default platforms based on the CPU
+architecture names in subdirectories called
+
+    /usr/lib/platform/<arch>-linux
+
+This is enough for regular Linux distributions, except Yocto. Yocto has
+so called Board Support Packages (BSPs) that cover more than the CPU
+architectures. These BSPs cover specific single board computer systems.
+Examples of such BSP names:
```
Comment (Conan-Kudo) [inline INSTALL:154]: Can you reword this to not require knowing what Yocto is? That is, if Yocto disappeared tomorrow, we wouldn't have a good reference for the need for this feature.

While we want to serve and support distributions, we try to ensure our features aren't "distro-dependant", so if the documentation on this can be made compelling without singling out or requiring to know Yocto, that would significantly help.

Comment (Conan-Kudo) [inline INSTALL:154]: Can you reword this to not require knowing what Yocto is? That is, if Yocto disappeared tomorrow, we wouldn't have a good reference for the need for this feature.

While we want to serve and support distributions, we try to ensure our features aren't "distro-dependant", so if the documentation on this can be made compelling without singling out or requiring to know Yocto, that would significantly help.

File: INSTALL [hunk 2]
```diff
@@ -143,6 +143,45 @@ and then install with:
...
+rpmbuild looks for is
+
+    /usr/lib/platform/`cat /etc/rpm/platform`-linux/macros
+
+If this file does not exist, many rpm macros will not have their expected
+values set and e.g. %configure will fail when trying to run rpmbuild.
+
+To allow creating the macros file for such a custom platform (only one
+needed for a specific BSP build in Yocto), the shell variables listed below
+must be set. If RPM_CUSTOM_ARCH is not set, the rest is ignored.
```
Comment (Conan-Kudo) [inline INSTALL:174]: Ditto here.

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

**Review by Conan-Kudo — APPROVED** [review]
This is weird, but I can't find anything overwhelmingly wrong with it

**Comment by ffesti** [comment]
Adjusted the docs a bit to not assume platform is always "linux" - even if it is for many people.
